File: solutions/php/language-list/1/LanguageList.php
<?php

declare(strict_types=1);

function language_list(string ...$languages): array
{
    return $languages;
}

function add_to_language_list(array $language_list, string $language): array {
    return [...$language_list, $language];
}

function prune_language_list(array $language_list): array {
    $copy = [...$language_list];
    array_shift($copy);
    return $copy;
}

function current_language(array $language_list): string {
    [$head] = $language_list;
    return $head;
}

function language_list_length(array $language_list): int {
    return  count($language_list);
}

File: solutions/elixir/language-list/1/lib/language_list.ex
defmodule LanguageList do
  def new() do
    []
  end

  def add(list, language) do
    [language | list]
  end

  def remove([_ | tail]) do
    tail
  end

  def first([head | _]) do
    head
  end

  def count([]) do
    0
  end

  def count([_ | tail]) do
    1 + count(tail)
  end

  def functional_list?([]) do
    false
  end

  def functional_list?(["Elixir" | _]) do
    true
  end

  def functional_list?([_ | tail]) do
    functional_list?(tail)
  end
end
